sysaudio/src/webaudio.zig

```diff
@@ -0,0 +1,108 @@
+const std = @import("std");
+const Mode = @import("main.zig").Mode;
+const DeviceDescriptor = @import("main.zig").DeviceDescriptor;
+const js = @import("sysjs");
+
+const Audio = @This();
+
+pub const Device = struct {
+    context: js.Object,
+
+    pub fn deinit(device: Device) void {
+        device.context.deinit();
+    }
+
+    pub fn pause(device: Device) void {
+        device.context.call("suspend", &.{});
+    }
+
+    pub fn start(device: Device) void {
+        _ = device.context.call("resume", &.{});
+    }
+};
+
+pub const DeviceIterator = struct {
+    ctx: *Audio,
+    mode: Mode,
+
+    pub fn next(_: DeviceIterator) IteratorError!?DeviceDescriptor {
+        return null;
+    }
+};
+
+pub const IteratorError = error{};
+
+pub const Error = error{
+    AudioUnsupported,
+};
+
+context_constructor: js.Function,
+
+pub fn init() Error!Audio {
+    const context = js.global().get("AudioContext");
+    if (context.is(.undef))
+        return error.AudioUnsupported;
+
+    return Audio{ .context_constructor = context.view(.func) };
+}
+
+pub fn deinit(audio: Audio) void {
+    audio.context_constructor.deinit();
+}
+
+pub fn waitEvents(_: Audio) void {}
+
+const default_channel_count = 2;
+const default_sample_rate = 48000;
+
+pub fn requestDevice(audio: Audio, config: DeviceDescriptor) Error!Device {
+    // NOTE: WebAudio only supports F32 audio format, so config.format is unused
+    const mode = config.mode orelse .output;
```

Comment:
shouldn't we require `mode` (like desktop implementation)? i don't think it's hard for user to just do `.{ .mode = .output }`. also has readibility benefit

Reply:
We should change `DeviceDescriptor`'s `mode` field:

```diff
-mode: ?Mode = null,
+mode: Mode = .output,
```